### ISSUE TITLE
php-fpm exploit

### ISSUE THREAD

**Issue by boralp** [digitalocean/nginxconfig.io#123]
I just found this exploit indexed and I believe nginxconfig.io users also can be effected.

https://github.com/neex/phuip-fpizdam

**Comment by atymic** [comment]
As per the list of preconditions I think configs are not affected, but would be good for someone to actually tests

> Nginx + php-fpm, location ~ [^/]\.php(/|$) must be forwarded to php-fpm (maybe the regexp can be stricter, see #1).

Our regex is stricter (but may still be vunerable)
```
location ~ \.php$ {
	include nginxconfig.io/php_fastcgi.conf;
}
```


> There must be a PATH_INFO variable assignment via statement fastcgi_param PATH_INFO $fastcgi_path_info;

Not present in the generated config

https://github.com/neex/phuip-fpizdam#the-full-list-of-preconditions

**Comment by 0xb4lint** [comment]
nginxconfig.io's generated configuration (`sites-available/example.com.conf`) by default is the following:
```nginx
location ~ \.php$ {
	include nginxconfig.io/php_fastcgi.conf;
}
```
*fix for preconditions `#1`*

In the `nginxconfig.io/php_fastcgi.conf` there is a line :
```nginx
try_files $fastcgi_script_name =404;
```
*fix for preconditions `#4`*

<br>
<br>
<br>

If you enable PHP legacy routing (https://nginxconfig.io/?0.php_legacy_routing), the generated config will be the following
```nginx
location ~ [^/]\.php(/|$) {
	# split path
	fastcgi_split_path_info ^(.+\.php)(/.+)$;
	set $_fastcgi_path_info $fastcgi_path_info;

	# 404
	try_files $fastcgi_script_name =404;

	...

	fastcgi_param PATH_INFO $_fastcgi_path_info;

	...
}
```
The `$_fastcgi_path_info` variable is required by issue https://trac.nginx.org/nginx/ticket/321.
So the `try_files` and `fastcgi_split_path_info` can work together. (*fix for preconditions `#4`*)

<br>
<br>
<br>

I've checked both the default and the legacy routing settings with `phuip-fpizdam` command and and the result were:
```
2019/10/30 10:50:45 Base status code is 404
2019/10/30 10:50:48 Detect() returned error: no qsl candidates found, invulnerable or something wrong
```
